src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataDeltaFeedSerializer.cs

```diff
@@ -441,12 +442,29 @@ public override async Task WriteObjectAsync(object graph, Type type, ODataMessag
         /// <param name="writeContext">The <see cref="ODataSerializerContext"/>.</param>
         public virtual void WriteDeltaDeletedEntry(object graph, ODataWriter writer, ODataSerializerContext writeContext)
         {
-            ODataDeletedResource deletedResource = GetDeletedResource(graph);
+            ODataResourceSerializer serializer = SerializerProvider.GetEdmTypeSerializer(_elementType) as ODataResourceSerializer;
 
-            if (deletedResource != null)
+            if (serializer != null)
+            {
+                ResourceContext resourceContext = serializer.GetResourceContext(graph, writeContext);
+                SelectExpandNode selectExpandNode = serializer.CreateSelectExpandNode(resourceContext);
+
+                if (selectExpandNode != null)
+                {
+                    ODataDeletedResource deletedResource = GetDeletedResource(graph, resourceContext, serializer, selectExpandNode, writeContext.IsUntyped);
+
+                    if (deletedResource != null)
```

Comment:
Based on the implementation of `GetDeletedResource` method, can `deletedResource` be null really or is this check superfluous?